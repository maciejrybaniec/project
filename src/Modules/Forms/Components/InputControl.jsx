/* @flow */
/**
 * Render input control component.
 * @module Modules/Forms/Components
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import autobind from 'autobind-decorator';
import React, { PureComponent } from 'react';

type InputControlType = 'text' | 'password';

type PropsType = {
    name: string,
    type: InputControlType,
    onChange: (name: string, value: string) => void
};

class InputControl extends PureComponent {
    /**
    * Set properties validation for component.
    */
    props: PropsType
    /**
    * Set default properties for component.
    */
    static defaultProps = {
        type: 'text'
    };
    constructor(props: PropsType) {
        super(props);
    }
    /**
     * Render component in DOM.
     * @returns {ReactElement}
     */
    render(): React.Element<*> {
        const { type, name } = this.props
        return (
            <input name={name} type={type} onChange={this._onChange} />
        );
    }
    /**
     * Change field value event handler.
     * @param {SyntheticEvent} event Synthetic event.
     * @private
     */
    @autobind
    _onChange(event: SyntheticEvent) {
        const { name, onChange } = this.props;
        if (event.target instanceof HTMLInputElement) {
            const fieldValue = event.target.value;
            onChange(name, fieldValue);
        }
    }
}

 export default InputControl;
