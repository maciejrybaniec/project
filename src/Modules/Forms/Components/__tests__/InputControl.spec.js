/**
 * Input control component specification.
 * @module Modules/Forms/Components/__tests__/InputControl
 * @copyright (c) 2016-present maciej.rybaniec@gmail.com
 */

import React from 'react';
import Renderer from 'react-test-renderer';
import sinon from 'sinon';
import { mount } from 'enzyme';

import InputControl from 'Modules/Forms/Components/InputControl';

describe('<InputControl />', () => {

    it('should set input control name', () => {
        const inputName = 'login';
        const wrapper = mount(<InputControl name={inputName} />);

        expect(wrapper.find(`input[name="${inputName}"]`).length).toEqual(1);
    });

    it('should set input control type', () => {
        const inputType = 'password';
        const wrapper = mount(<InputControl type={inputType} />);

        expect(wrapper.find(`input[type="${inputType}"]`).length).toEqual(1);
    });

    it('should trigger callback on change', () => {
        const inputName = 'login';
        const spy = sinon.spy();

        const wrapper = mount(
            <InputControl
                onChange={spy}
                name={inputName}
            />
        );

        const input = wrapper.find('input');
        input.node.value = 'login';
        input.simulate('change', input);

        expect(spy.calledOnce).toBe(true);
    });

});
