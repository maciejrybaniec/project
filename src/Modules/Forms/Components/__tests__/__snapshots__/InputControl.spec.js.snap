exports[`<InputControl /> should render <InputControl /> component correctly 1`] = `
<input
  name="login"
  onChange={[Function]}
  type="text" />
`;
